lib/Dialect/Torch/Transforms/DecomposeComplexOps.cpp

```diff
@@ -384,6 +384,259 @@ class DecomposeAtenReshapeOp : public OpRewritePattern<AtenReshapeOp> {
 };
 } // namespace
 
+namespace {
+// support static shape only
+class DecomposeAtenEinsumOp : public OpRewritePattern<AtenEinsumOp> {
+public:
+  using OpRewritePattern::OpRewritePattern;
+  LogicalResult matchAndRewrite(AtenEinsumOp op,
+                                PatternRewriter &rewriter) const override {
+    Location loc = op.getLoc();
+    std::string equation;
+    if (!matchPattern(op.getEquation(), m_TorchConstantStr(equation))) {
+      return rewriter.notifyMatchFailure(op, "Unsupported value of equation");
+    }
+
+    SmallVector<SmallVector<char>> inputTokens;
+    SmallVector<char> resultTokens;
+    SmallVector<char> inputToken;
+    SmallVector<SmallVector<Value>> inputShapes;
+
+    size_t index = 0;
+    enum EquationVariable { kIsInput, kIsResult };
+    EquationVariable currentVariable = kIsInput;
+    while (index < equation.size()) {
+      if (std::isalpha(equation[index])) {
+        if (currentVariable == kIsInput) {
+          inputToken.push_back(equation[index]);
+        } else {
+          resultTokens.push_back(equation[index]);
+        }
+      } else if (equation.substr(index, 1).find(",") != std::string::npos) {
+        inputTokens.emplace_back(std::move(inputToken));
+      } else if ((index < (equation.size() - 1)) &&
+                 (equation.substr(index, 2).find("->") != std::string::npos)) {
+        inputTokens.emplace_back(std::move(inputToken));
+        currentVariable = kIsResult;
+        index++;
+      } else {
+        return rewriter.notifyMatchFailure(op, "unexpected character encountered");
+      }
+      index++;
+    }
+
+    SmallVector<Value> torchTensors;
+    if (!getListConstructElements(op.getTensors(), torchTensors)) {
+      return rewriter.notifyMatchFailure(
+          op, "input should comes from a PrimListConstructOp");
+    }
+
+    int64_t totalRank = torchTensors.size();
+    for (int i = 0; i < totalRank; i++) {
+      BaseTensorType tensorType = torchTensors[i].getType().cast<BaseTensorType>();
+      ArrayRef<int64_t> inputShape = tensorType.getSizes();
```

Comment:
There is no guarantee that tensors will have sizes at this point, so you need to check `tensorType.hasSizes()` first